Libraries/src/net/relinc/libraries/application/StrainGauge.java

```diff
@@ -27,11 +29,43 @@ public StrainGauge(String genName, double GaugeFactor, double Resistance, double
 		length = Length;
 		voltageCalibrated = VoltageCalibrated;
 	}
-	public StrainGauge(String path){
-		String file = SPOperations.readStringFromFile(path);
-		setParametersFromString(file);
+	public StrainGauge(String path)  {
+		String file_contents = SPOperations.readStringFromFile(path);
+		if(path.contains(".json")) {
+			this.setParametersFromJSONString(file_contents);
+		} else {
+			setParametersFromString(file_contents);
+		}
 	}
-	
+	public StrainGauge(StrainGauge copy) {
+		genericName = copy.genericName;
+		gaugeFactor = copy.gaugeFactor;
+		resistance = copy.resistance;
+		shuntResistance = copy.shuntResistance;
+		length = copy.length;
+		voltageCalibrated = copy.voltageCalibrated;
+	}
+	public void setParametersFromJSONString(String file) {
+
+		JSONParser jsonParser = new JSONParser();
+		JSONObject jsonObject = null;
```

Comment:
try to avoid `null`. Perhaps put it inside the `try`, perhaps a Java `Maybe` is a good option